### PR TITLE
More usable timeline (fixes #15 fixes #155 connects to #27)

### LINKED ISSUE #15: Fix timeline for huge amount of hosts

### LINKED ISSUE #155: Fix timeline scroll & height
Empty screen, no brush, no scroll

screen shot 2017-02-14 at 18 48 06

### PR DESCRIPTION
**Pull request by sseliverstov**
* add minimum duration filter (reduce count)
* stick zoom bar on bottom for tall graph
* rounded bars for tests (for visually split)

### CODE REVIEW

File: allure-web/src/plugins/timeline/TimelineModel.js [hunk 1]
```diff
@@ -14,4 +14,40 @@ export default class TimelineModel extends Model {
...
+                    return Object.assign({}, thread, { children: thread['children'].filter(testCase => {
+                            if (testCase.time.duration >= minDuration){
+                                return true;
+                            } else {
+                                total--;
+                                return false;
+                            }
+                        })
+                    });
+                }).filter(d=> {return d['children'].length;})});
```
Comment (just-boris) [inline allure-web/src/plugins/timeline/TimelineModel.js]: `d.children.length`?
String keys are used only for dynamic lookup, like `d[keyName]`

File: allure-web/src/plugins/timeline/TimelineModel.js [hunk 2]
```diff
@@ -14,4 +14,40 @@ export default class TimelineModel extends Model {
...
+                };
+            })),
+            []
+        );
+    }
+
+    getFilteredData(minDuration){
+        var data = this.get('children');
+        var total = this.get('statistic').total;
+        this.data = Object.assign({}, {
```
Comment (just-boris) [inline allure-web/src/plugins/timeline/TimelineModel.js]: `Object.assign` does nothing here.
You can just do
```js
return {
   children: data.map(....),
   selectedTestCases: this.get('statistic').total
}
```

File: allure-web/src/plugins/timeline/TimelineView.js [hunk 3]
```diff
@@ -46,72 +45,115 @@ class TimelineView extends BaseChartView {
...
-    doShow() {
-        const width = this.$el.width() > 2 * PADDING ? this.$el.width() - 2 * PADDING : this.$el.width();
-
-        const [from, to] = this.model.getAllTestcases().reduce(
-            ([from, to], {time}) => [Math.min(from, time.start), Math.max(to, time.stop)],
-            [Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY]
-        );
+    setupViewport() {
+        var selectedPercent = (100*this.data.selectedTestCases/this.model.get('statistic').total).toFixed(2);
+        this.$el.html(`<div class="timeline__header">
```
Comment (just-boris) [inline allure-web/src/plugins/timeline/TimelineView.js]: It's better to place so much HTML into hbs file.

File: allure-web/src/plugins/timeline/styles.css [hunk 4]
```diff
@@ -2,40 +2,169 @@
...
   &__left_handle {
     text-anchor: end;
   }
+
 }
 
-.selection {
-  fill: steelblue;
-  fill-opacity: 0.15;
+input[type=range] {
```
Comment (just-boris) [inline allure-web/src/plugins/timeline/styles.css]: These styles will be applied to **all** inputs that we have in Allure face.
Use class name selector `.timeline__range_bar`

File: allure-web/src/plugins/timeline/TimelineView.js [hunk 5]
```diff
@@ -47,45 +48,28 @@ class TimelineView extends BaseChartView {
...
-                    </rect>
-                    <g class="timeline__brush__axis timeline__brush__axis_x"></g>
-                </svg>
-            </div>
-        </div>`);
+        this.$el.html(template({
+            minDuration: this.minDuration,
+            selectedTestCases: this.data.selectedTestCases,
+            selectedPercent: selectedPercent,
+            selectedDuration: this.selectedDuration|0,
```
Comment (just-boris) [inline allure-web/src/plugins/timeline/TimelineView.js]: This is bitwise operator. Did you mean `selectedDuration: this.selectedDuration || 0,`

File: allure-web/src/plugins/timeline/TimelineModel.js [hunk 6]
```diff
@@ -14,4 +14,40 @@ export default class TimelineModel extends Model {
...
+                };
+            })),
+            []
+        );
+    }
+
+    getFilteredData(minDuration){
+        var data = this.get('children');
+        var total = this.get('statistic').total;
+        this.data = {
```
Comment (just-boris) [inline allure-web/src/plugins/timeline/TimelineModel.js]: you can put `return` right here. No need to store value into `this.data`

File: allure-web/src/plugins/timeline/TimelineView.js (hunk 5 above)
Comment (sseliverstov) [inline allure-web/src/plugins/timeline/TimelineView.js]: float to int

Comment (just-boris) [inline allure-web/src/plugins/timeline/TimelineView.js]: Then, let's make it clear.
`Math.floor(this.selectedDuration)`

**Pull request merged**